tracer/src/Datadog.Trace/TraceProcessors/TraceUtil.cs

```diff
@@ -0,0 +1,344 @@
+// <copyright file="TraceUtil.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.TraceProcessors
+{
+    internal class TraceUtil
+    {
+        // MaxTagLength the maximum length a tag can have
+        private const int MaxTagLength = 200;
+
+        private static readonly UTF8Encoding Encoding = new UTF8Encoding(false);
+
+        // https://github.com/DataDog/datadog-agent/blob/0454961e636342c9fbab9e561e6346ae804679a9/pkg/trace/traceutil/truncate.go#L34-L49
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool TruncateUTF8(ref string value, int limit)
+        {
+            if (string.IsNullOrEmpty(value) || Encoding.GetByteCount(value) <= limit)
+            {
+                return false;
+            }
+
+            var valueCharArray = value.ToCharArray();
+            for (var i = 0; i < valueCharArray.Length; i++)
+            {
+                if (Encoding.GetByteCount(valueCharArray, 0, i) > limit)
```

Comment:
We would need to benchmark to validate this, but my guess is that calling `Encoding.GetByteCount()` on progressively longer strings will be very slow for long SQL queries. It might be better to get the size in bytes of each character and keep adding as you go until you reach the limit.

Alternatively, this doesn't need to be perfect and maybe assuming 1 byte per character is a good estimate and we can just truncate the string at 200 characters. I think if we send something a little longer it will be truncated later, anyway.

Reply:
Resolving this comment since it's not part of your changes.